libs/blocks/carousel/carousel.js

```diff
@@ -1 +1,381 @@
+    if (swipeDistance.xDistance > swipe.xMin) {
+      moveSlides(event, carouselElements);
+    }
+  });
+  /* c8 ignore end */
+}
+
+function handleChangingSlides(carouselElements) {
+  const { el, nextPreviousBtns, slideIndicators } = carouselElements;
+
+  // Handle Next/Previous Buttons
+  [...nextPreviousBtns].forEach((btn) => {
+    btn.addEventListener('click', (event) => {
+      moveSlides(event, carouselElements);
+    });
+  });
+
+  // Handle keyboard navigation
+  el.addEventListener('keydown', (event) => {
+    if (event.key === KEY_CODES.ARROW_RIGHT
+      || event.key === KEY_CODES.ARROW_LEFT) { moveSlides(event, carouselElements); }
+  });
+
+  // Handle slide indictors
+  [...slideIndicators].forEach((li) => {
+    li.addEventListener('click', (event) => {
+      const jumpToIndex = Number(li.dataset.index);
+      moveSlides(event, carouselElements, jumpToIndex);
+    });
+  });
+
+  // Swipe Events
+  mobileSwipeDetect(carouselElements);
+}
+
+export default function init(el) {
+  const carouselSection = el.closest('.section');
+  if (!carouselSection) return;
+
+  const keyDivs = el.querySelectorAll(':scope > div > div:first-child');
+  const carouselName = keyDivs[0].textContent;
+  const parentArea = el.closest('.fragment') || document;
+  const candidateKeys = parentArea.querySelectorAll('div.section-metadata > div > div:first-child');
+  const slides = [...candidateKeys].reduce((rdx, key) => {
+    if (key.textContent === 'carousel' && key.nextElementSibling.textContent === carouselName) {
+      const slide = key.closest('.section');
+      slide.classList.add('carousel-slide');
+      rdx.push(slide);
+      slide.setAttribute('data-index', rdx.indexOf(slide));
+    }
+    return rdx;
+  }, []);
+
+  const fragment = new DocumentFragment();
+  const nextPreviousBtns = decorateNextPreviousBtns();
+  const slideIndicators = decorateSlideIndicators(slides);
+  const controlsContainer = createTag('div', { class: 'carousel-controls' });
+
+  fragment.append(...slides);
+  const slideWrapper = createTag('div', { class: 'carousel-wrapper' });
+  const slideContainer = createTag('div', { class: 'carousel-slides' }, fragment);
+  const carouselElements = {
+    el,
+    nextPreviousBtns,
+    slideContainer,
+    slides,
+    slideIndicators,
+    controlsContainer,
+    direction: undefined,
+  };
+
+  if (el.classList.contains('lightbox')) {
+    const lightboxBtns = decorateLightboxButtons();
+    slideWrapper.append(slideContainer, ...lightboxBtns);
+    handleLightboxButtons(lightboxBtns, el, slideWrapper);
+  } else {
+    slideWrapper.append(slideContainer);
+  }
+
+  el.textContent = '';
+  el.append(slideWrapper);
+
+  const dotsUl = createTag('ul', {
+    class: 'carousel-indicators',
+    role: 'tablist',
+    tabindex: 0,
+  });
+  dotsUl.append(...slideIndicators);
+  controlsContainer.append(dotsUl);
+
+  el.append(...nextPreviousBtns, controlsContainer);
+
+  parentArea.addEventListener('milo:deferred', () => {
```

Comment:
Should the event listener be removed after `milo:deferred` has fired?

Reply:
That event only fires once per page. 

cc: @auniverseaway

Reply:
may as well remove the event listener in that case

Reply:
Done.